Closes #19865: Improve CableTypeChoices structure and grouping

pheus opened the pull request:
<!-- Thank you for your interest in contributing to NetBox! Please note that our contribution policy requires that a feature request or bug report be approved and assigned prior to opening a pull request. This helps avoid wasted time and effort on a proposed change that we might not be able to accept. IF YOUR PULL REQUEST DOES NOT REFERENCE AN ISSUE WHICH HAS BEEN ASSIGNED TO YOU, IT WILL BE CLOSED AUTOMATICALLY. Please specify your assigned issue number on the line below. -->

### Fixes: #19865

### Summary
Regroups `CableTypeChoices` into clearer headings to improve scanability of the cable type dropdown. This is a UI‑only change; no slugs or stored values are modified.

### Proposed UI grouping

- **Copper — Ethernet (Twisted Pair)**
  - CAT3, CAT5, CAT5e, CAT6, CAT6a, CAT7, CAT7a, CAT8, MRJ21 Trunk
- **Copper — Twinax (DAC)**
  - Direct Attach Copper (Active), Direct Attach Copper (Passive)
- **Copper — Coaxial**
  - Coaxial
- **Fiber — Multimode**
  - Multimode Fiber (OM1/OM2/OM3/OM4/OM5)
- **Fiber — Singlemode**
  - Singlemode Fiber (OS1/OS2)
- **Fiber — Other / Active**
  - Active Optical Cabling (AOC)
- **Other**
  - USB, Power

### Rationale
Makes the dropdown easier to read by grouping related families; keeps coax generic for now.

### Backward compatibility
No breaking changes. Existing API slugs and database values remain unchanged.

jeremystretch reviewed (changes requested):
Thanks @pheus! I just have a couple editorial tweaks.

pheus commented on netbox/dcim/choices.py:
```diff
@@ -1535,28 +1551,57 @@ class CableTypeChoices(ChoiceSet):
         ),
         (
-            _('Fiber'), (
+            _('Fiber - Multi Mode'),
```
Will do, Thanks!

pheus commented on netbox/dcim/choices.py:
```diff
@@ -1507,26 +1508,41 @@ class CableTypeChoices(ChoiceSet):
     CHOICES = (
         (
-            _('Copper'), (
+            _('Copper - Twisted Pair (Ethernet)'),
```
TBH, I was thinking about the same, but didn't come up with "UTP/STP". I will update it. Thanks!

pheus committed:
feat(dcim): Improve CableTypeChoices structure and grouping

Refactors `CableTypeChoices` by reorganizing cable types into more
specific subcategories. Enhances clarity with distinct groups such as
Copper (Twisted Pair, Twinax, Coaxial) and Fiber (Multi Mode, Single
Mode, Other).

Closes #19865

jeremystretch reviewed (approved):
Thanks @pheus!
